### PR TITLE
feat(@angular/cli): add scope hoisting via webpack 3

### PR DESCRIPTION
**Pull request by filipesilva**
This should result in significant bundle size reduction. For best results, build for production with `--vendor-chunk=false`.

See https://medium.com/webpack/webpack-3-official-release-15fd2dd8f07b for details.

---

Blockers for this to get merged:
- [x] Webpack 3 released (see https://medium.com/webpack/webpack-3-official-release-15fd2dd8f07b)
- [x] ~Webpack 2.5.1+ bug with CommonsChunkPlugin https://github.com/webpack/webpack/issues/4850 fixed, it's breaking `tests/e2e/tests/misc/common-async.ts`~ (not a bug, just config https://github.com/webpack/webpack/issues/5109#issuecomment-311154920)
- [x] Webpack 2.4.x+ build performance regression https://github.com/webpack/webpack/issues/4863
- [x] ~`extract-text-webpack-plugin` generates css in the wrong order https://github.com/webpack-contrib/extract-text-webpack-plugin/issues/548~ (not applicable to CLI projects https://github.com/webpack-contrib/extract-text-webpack-plugin/issues/548#issuecomment-311326232)
- [x] Packages allowing `webpack@3.0.0` as a peer dependency:
  - [x] `extract-text-webpack-plugin`: https://github.com/webpack-contrib/extract-text-webpack-plugin/pull/540
  - [x] `html-webpack-plugin`: https://github.com/jantimon/html-webpack-plugin/pull/707
  - [x] `less-loader`: https://github.com/webpack-contrib/less-loader/pull/208
  - [x] `webpack-dev-middleware`: https://github.com/webpack/webpack-dev-middleware/pull/200
  - [x] `webpack-dev-server`: https://github.com/webpack/webpack-dev-server/pull/946

### CODE REVIEW

File: packages/@angular/cli/models/webpack-configs/common.ts [hunk 1]
```diff
@@ -97,11 +97,11 @@ export function getCommonConfig(wco: WebpackConfigOptions) {
     ].concat(extraPlugins),
     node: {
       fs: 'empty',
-      global: true,
+      global: 'empty',
```
Comment (clydin) [inline packages/@angular/cli/models/webpack-configs/common.ts]: global only supports a boolean.  Ref: https://webpack.js.org/configuration/node/#node-global

**Comment by intellix** [comment]
:shipit: It seems they released Scope hoisting and blog posted about it

**Comment by filipesilva** [comment]
Pushed an update that uses `webpack@3.0.0`, but there seem to be some performance regressions atm: https://github.com/webpack/webpack/issues/4863.

**Comment by TheLarkInn** [comment]
We have a few PRs we are hoping to land within this next few weeks. It should good the perf regressions, in theory.

**Comment by d3viant0ne** [comment]
@filipesilva - I'll see that the `webpack-contrib` related repos sorted this weekend.

**Comment by d3viant0ne** [comment]
@filipesilva - You can check less-loader off your list https://github.com/webpack-contrib/less-loader/releases/tag/v4.0.5

Extract Text will make the `dist-tag` tomorrow evening.

**Comment by filipesilva** [comment]
@d3viant0ne thank you for the heads up, I just updated this PR to use `less-loader`. For now it's still using `extract-text-webpack-plugin@3.0.0-rc.2` but will be updated with `3.0.0` once it is released.

**Comment by filipesilva** [comment]
This PR no longer automatically turn off vendor chunking in production builds, since it's benefits might be conditional on a bug that needs to be fixed. This can be added at a future date if needed.

**Comment by filipesilva** [comment]
Updated to `extract-text-webpack-plugin@3.0.0`, which was the last missing piece. 

I'd like to extend my thanks to @d3viant0ne and @TheLarkInn, they've been awesome in communicating changes and releases in the Webpack 3 rollout, cheers!

**Pull request merged**